processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go

```diff
@@ -71,5 +82,72 @@ func (d *Detector) Detect(ctx context.Context) (pdata.Resource, error) {
 	attr.InsertString(conventions.AttributeHostType, meta.InstanceType)
 	attr.InsertString(conventions.AttributeHostName, hostname)
 
+	tags, err := connectAndFetchEc2Tags(meta.Region, meta.InstanceID, d.cfg)
+	if err != nil {
+		return res, fmt.Errorf("failed fetching ec2 instance tags: %w", err)
+	}
+
+	for key, val := range tags {
+		attr.InsertString(tagPrefix+key, val)
+	}
+
 	return res, nil
 }
+
+func connectAndFetchEc2Tags(region string, instanceID string, cfg Config) (map[string]string, error) {
+	if len(cfg.Tags) == 0 {
+		return nil, nil
+	}
+
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(region)},
+	)
+	if err != nil {
+		return nil, err
+	}
+	e := ec2.New(sess)
+
+	return fetchEC2Tags(e, instanceID, cfg)
+}
+
+func fetchEC2Tags(svc ec2iface.EC2API, instanceID string, cfg Config) (map[string]string, error) {
+	ec2Tags, err := svc.DescribeTags(&ec2.DescribeTagsInput{
+		Filters: []*ec2.Filter{{
+			Name: aws.String("resource-id"),
+			Values: []*string{
+				aws.String(instanceID),
+			},
+		}},
+	})
+	if err != nil {
+		return nil, err
+	}
+	tags := make(map[string]string)
+	for _, tag := range ec2Tags.Tags {
+		matched := regexArrayMatch(cfg.Tags, *tag.Key)
+		if matched {
+			tags[*tag.Key] = *tag.Value
+		}
+	}
+	return tags, nil
+}
+
+func validateRegexes(cfg Config) error {
+	for _, elem := range cfg.Tags {
+		_, err := regexp.Compile(elem)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func regexArrayMatch(arr []string, val string) bool {
+	for _, elem := range arr {
+		matched, _ := regexp.MatchString(elem, val)
```

Comment:
I was not very clear in my previous comment, let me try to rewrite it in more detail. The usual way to work with regexes in Go is to [`Compile`](https://golang.org/pkg/regexp/#Compile) them and use the [`MatchString` method](https://golang.org/pkg/regexp/#Regexp.MatchString) from the resulting `Regexp` struct (this is more efficient since you only `Compile` once and it's cleaner since you don't have to ignore errors that you know won't happen).

 What I meant here thus is to

1. create a `[]*regexp.Regexp` array when creating the `Detector` by compiling each regex (and failing if any fails)
2. in `regexArrayMatch` you would pass `regexArrayMatch(arr []*regexp.Regexp, val string)` and apply the `Regexp.MatchString` method for each element of `arr`. This method only returns a `bool` so you don't have to ignore the error like you are doing now.

Reply:
Thanks very much for the detailed explanation, that makes a lot of sense, will make the change